dropwizard-metrics5/src/main/java/feign/metrics5/MeteredDecoder.java

```diff
@@ -63,14 +64,14 @@ public Object decode(Response response, Type type)
     } catch (IOException | RuntimeException e) {
       metricRegistry.meter(
           metricName.metricName(template.methodMetadata(), template.feignTarget(), "error_count")
-              .tagged("exception_name", e.getClass().getSimpleName())
+              .tagged("exception_name", ExceptionUtils.getRootCause(e).getClass().getSimpleName())
```

Comment:
I'm thinking a bit more about this.... we are breaking the existing contract with this change.

Can we leave `exception_name` as was and add `root_cause_name` for this?

Cheers

Reply:
yes, that makes more sense, made the same change in https://github.com/OpenFeign/feign/pull/1883/commits/23cb5ed45d71937dc61d5214861997f99cbe561b ,  can you take a look at it again?